rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -267,6 +267,7 @@ void RecorderImpl::record()
     throw std::runtime_error("No serialization format specified!");
   }
 
+  subscriptions_.clear();
```

Comment:
Is this worth adding a test for?

Reply:
I don't think so. It is more like defensive programming. No real issue so far.